pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -190,7 +190,7 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			},
 			DHCPDeactivatedOnSubnet: !enableDHCP,
 		}
-		if !enableDHCP && len(nsxSubnetPortState.RealizedBindings) > 0 {
+		if len(nsxSubnetPortState.RealizedBindings) > 0 {
```

Comment:
I recall we add this for another bug. Without checking the dhcp mode, we will set the ip in status for a dhcp subnetport without addressbinding when operator is restarted, which is not expected.
Shall we only update the CR status for the case that the spec.addressbinding is provided?

Reply:
Good point! Fixed, thanks!